public/app/core/utils/richHistory.ts

```diff
@@ -304,3 +334,17 @@ export function createDatasourcesList(queriesDatasources: string[]) {
   });
   return datasources;
 }
+
+export function notEmptyQuery(query: DataQuery) {
```

Comment:
I think in the same way we will need to handle the case query is still a string.